frontend/src/Components/App.tsx

```diff
@@ -0,0 +1,195 @@
+import React from "react";
+import {useState, useEffect} from "react";
+
+import { 
+  Card, 
+  CardActionArea, 
+  CardActions, 
+  CardContent, 
+  CardMedia, 
+  Button, 
+  Typography,
+  Container,
+  Paper,
+  CssBaseline,
+  Grid
+} from '@material-ui/core';
+
+import Error from "./Error";
+import Loading from "./Loading";
+import Product from "../Interfaces/Product";
+import PopupBox from "./PopupBox";
+
+import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCartOutlined';
+import DeleteForeverIcon from '@material-ui/icons/DeleteForeverOutlined';
+import "../Styles/App.css"
+import APIService from '../APIService'
+import PopupBoxProps from "../Interfaces/PopupBoxProps";
+import Cookies from "../Cookies"
+
+export default function App(){
+
+	const [error, setError] = useState<string>("");
+	const [isLoaded, setIsLoaded] = useState<boolean>(false);
+	const [items, setItems] = useState<Product[]>([]);
+	const [cart, setCart] = useState<number[]>([]);
+	const [popupProps, setPopupProps] = useState<PopupBoxProps | undefined>();
+
+	function updateStorageCart(){
```

Comment:
Functions within an React functional components should be wrapped with a `useCallback` hook to avoid performance issue, as plain functions will be generated again upon every single render